prov/verbs/src/verbs_cq.c

```diff
@@ -586,6 +586,13 @@ int fi_ibv_cq_open(struct fid_domain *domain, struct fi_cq_attr *attr,
 
 	_cq->domain = container_of(domain, struct fi_ibv_domain,
 				   util_domain.domain_fid);
+	if (_cq->domain->info->domain_attr->threading != FI_THREAD_SAFE) {
+		_cq->cq_fastlock_acquire = fi_ibv_fastlock_empty;
+		_cq->cq_fastlock_release = fi_ibv_fastlock_empty;
```

Comment:
I'm not sure no-op can be used in all cases.  THREAD_DOMAIN or THREAD_COMPLETION seem possible.  THREAD_FID or THREAD_ENDPOINT may still require locking.

Reply:
Yep, I agree with that. The access to CQ resource (lsit and verbs CQ) must be protected in case of FID, SAFE, ENDPOINT threading levels.